dataeng/resources/warehouse-transforms-ci-dbt.sh

```diff
@@ -11,4 +11,24 @@ dbt deps --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --prof
 dbt seed --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
 
 dbt run $DBT_RUN_OPTIONS $DBT_RUN_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
-dbt test $DBT_TEST_OPTIONS $DBT_TEST_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET
\ No newline at end of file
+dbt test $DBT_TEST_OPTIONS $DBT_TEST_EXCLUDE --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/ --profile $DBT_PROFILE --target $DBT_TARGET && true
+if [ $? -eq 1 ]
+then
+    if [ "$WITH_RETRY" == "true" ]
+    then
+        pip install -r ../../tools/ci_scripts/requirements.txt
+        if [ "$DBT_TEST_EXCLUDE" == "" ]
+        then
+            python ../../tools/ci_scripts/rerun_flaky_tests.py --project-path . --profiles-dir $WORKSPACE/analytics-secure/warehouse-transforms/  --profile $DBT_PROFILE  \
+            --target $DBT_TARGET --count $NO_OF_TRIES
+        else
+            PREFIX="--exclude "
+            TEST_EXCLUSIONS=$(echo "$DBT_TEST_EXCLUDE" | sed -e "s/^$PREFIX//")
```

Comment:
nit: I feel like this is more complicated than needed. In general, I prefer explicit values over manipulated ones. I understand why you did it this way, but I just wonder if there is a simpler way of passing this value. Perhaps something like:
```
DBT_EXCLUSIONS="x,y,z"
DBT_EXCLUSION_STRING="--exclude $DBT_EXCLUSIONS"
```
that way, you have both formats. Maybe we just need to change the name of the parameter in the python script

At the end of the day, it's not really that important, and you should be able to merge this, but if you have a spare moment, I feel like we can make this more clear.